File: dart/lib/bebop_handlers/ready.dart
import 'bebop_handler.dart';
import './player_joined.dart';
import 'package:lean_dart_client/schemas.g.dart';

class ReadyMsgHandler extends BebopHandler {
  @override
  void handleMessage(sender, message) {
    message = ReadyMsg.decode(message);
    print(
        '${PlayerJoinedMsgHandler.opponentName} is ready. Game will start soon.');
  }
}

File: typescript/src/bebop_handlers/ready.ts
import { ReadyMsg } from '../models/pactheman.models';
import BebopHandler from './bebopHandler';
import PlayerJoinedMsgHandler from './playerJoined';

export default class ReadyMsgHandler implements BebopHandler {
    handleMessage(sender: any, message: any): void {
        message = ReadyMsg.decode(message);
        console.log(`${PlayerJoinedMsgHandler.opponentName} is ready. Game will start soon.`);
    }
}